barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp

```diff
@@ -148,9 +148,7 @@ void ClientIVC::complete_kernel_circuit_logic(ClientCircuit& circuit)
  */
 void ClientIVC::accumulate(ClientCircuit& circuit, const std::shared_ptr<VerificationKey>& precomputed_vk, bool mock_vk)
 {
-    is_kernel = !is_kernel; // toggle on each call (every even circuit is a kernel)
```

Comment:
This is the main point of this PR. IVC will non longer internally assume that every other circuit is a kernel. Instead it will check `circuit.databus_propagation_data.is_kernel`. This tag will be set to true automatically in ACIR land when the corresponding noir program uses calldata (since only kernels use calldata).